File: practicetools/albertadrugprice/Backup/archive/05 - Web Files/live_search.php
<?php
//studybuf_abc_vw	bbZMZ*Fl%[1ANQqFM~
//get the q parameter from URL
$q=$_GET["q"];

//Establish database connection
$conn = mysqli_connect('localhost','studybuf_dblview','iDBL1403#sb','studybuf_abc_dbl');
if (!$conn)
{
    die('Could not connect: ' . mysqli_error($con));
}

$sql="SELECT `BrandName`, `GenericName` FROM dbl WHERE (BrandName LIKE '%" . $q . "%' OR GenericName LIKE '%" . $q . "%')";
$result = mysqli_query($conn, $sql);

//
//lookup all links from the xml file if length of q>0
if (strlen($q) > 0)
{
	$entries = "";
	
	$ResultArray = array();
	
	//Create a list out of results
	while($row = mysqli_fetch_array($result))
	{
		$TempArray = array("BrandName"=>$row['BrandName'], "GenericName"=>$row['GenericName']);
		array_push($ResultArray, $TempArray);
	}
	
	//Removes entries from array that are duplicate brand names
	$TempArray1 = array();
	
	foreach ($ResultArray as &$InnerArray)
	{
		if (!isset($TempArray1[$InnerArray['BrandName']]))
		{
			$TempArray1[$InnerArray['BrandName']] = &$InnerArray;
		}
	}
	
	//Converts the final unique array back to original array
	$ResultArray = array_values($TempArray1);
	
	//For each group of generic names, combines the brand names
	$TempArray2 = Array();

	foreach ($ResultArray as $key => $InnerArray)
	{
		if ($key == 0)
		{
			$TempArray2[0] = $InnerArray;
		}
		else
		{
			$match = FALSE;
			for ($i = 0; $i < count($TempArray2); $i++)
			{
				if ($TempArray2[$i]['GenericName'] == $InnerArray['GenericName'])
				{
					$TempArray2[$i]['BrandName'] = $TempArray2[$i]['BrandName'] . ", " . $InnerArray['BrandName'];
					$match = TRUE;
					break;
				}
			}
			
			if ($match == FALSE)
			{
				array_push($TempArray2, $InnerArray);
			}
		}
	}
	
	//Creates list for return via AJAX function
	foreach ($TempArray2 as &$InnerArray)
	{
		$entries .= "<li><a id='" . str_replace(" ", "_", $InnerArray['GenericName']) . "' onclick='ChooseResult(this.id)'><b>" . $InnerArray['GenericName'] . "</b><br><i>also known as " . $InnerArray['BrandName'] . "</a></i>";
	}
}

// Set output to "no suggestion" if no hint was found or to the correct values
if ($entries=="")
{
	$response="<ul><li><a><b>No medication found</b></a></ul>";
}
else
{
	$response = "<ul>" . $entries . "</ul>";
}

//output the response
echo $response;
?>

File: sb-django/drug_price_calculator/static/js/live_search.py
"""
    <?php

    require_once '../../../../config/db_config.php';

    $response = "";

    //get the q parameter from URL
    $q = $_GET["q"];

    // Only run search if $q is not blank
    if (strlen($q) > 0) {
	    // Adding wildcard markers to query
	    $q = "%" . $q . "%";
	
	    // Establish database connection
	    $db = db_connect('sb', 'abc_vw');
	
	    // Sending prepared statement to server
	    $query = "SELECT DISTINCT t1.url , t1.brand_name, t1.strength, " . 
			     "t1.route, t1.dosage_form, t1.generic_name " . 
			     "FROM abc_price t1 " . 
			     "INNER JOIN abc_price t2 " .
			     "ON t1.generic_name = t2.generic_name " .
			     "WHERE ((t2.generic_name LIKE ? OR t2.brand_name LIKE ?) " . 
			     "AND t1.unit_price IS NOT NULL)";
	
	    $resultArray = returnResults($db, $query, [$q, $q]);
	
	    // If results were obtained, finish processing
	    if (count($resultArray) > 0) {
		    // For each group of generic names + strength + route + 
		    // dosage form, combines the brand names
		    $tempArray = array();
		
		    foreach ($resultArray as $key => $item) {
			    $tempText = $item['generic_name'] . " (" . $item['strength'] . " " . 
						    $item['route'] . " " . $item['dosage_form'] . ")";
						
			    //Cleans up the tempText in case an item was missing
			    $tempText = str_replace("  ", " ", $tempText);
			    $tempText = str_replace("  ", " ", $tempText);
			    $tempText = str_replace("( ", "(", $tempText);
			    $tempText = str_replace(" )", ")", $tempText);
			    $tempText = str_replace(" ()", "", $tempText);
			
			    if ($key == 0) {
				    $tempArray[0] = array('title' => $tempText,
									      'url' => $item['url'],
									      'brand_name' => $item['brand_name']);
			    } else {
				    $match = FALSE;
				
				    for ($i = 0; $i < count($tempArray); $i++) {
					    if ($tempArray[$i]['title'] == $tempText) {
						    $tempArray[$i]['url'] .= "," . $item['url'];
						    $tempArray[$i]['brand_name'] .= ", " . $item['brand_name'];
						    $match = TRUE;
						    break;
					    }
				    }
				
				    if ($match == FALSE) {
					    $index = count($tempArray);
					    $tempArray[$index] = array('title' => $tempText,
											       'url' => $item['url'],
											       'brand_name' => $item['brand_name']);
				    }
			    }
		    }

		    $resultArray = array_values($tempArray);
		    $entries = "";
		
		    //Creates list for return via AJAX function
		    foreach ($resultArray as $index => $item) {
			    $entries .= "<li><a id='Search-Result-" . $index . "' " .
						    "data-url='" . $item['url'] . "' " .
						
						    "onclick='chooseResult(this)'>" .
						    "<strong>" . $item['title'] . "</strong><br>" .
						    "<em>also known as " . $item['brand_name'] . "</em></a></li>";
		    }
		
		    $response = "<ul>" . $entries . "</ul>";
	    } else {
		    $response = "<ul><li><a><strong>No medication found</strong></a></li></ul>";
	    }
    }

    // Output the response
    echo $response;
    ?>
"""